webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/domain/queries/NodeQueryCriteriaData.scala

```diff
@@ -519,6 +523,50 @@ final case class NodeCriterionMatcherString(extractor: CoreNodeFact => Chunk[Str
   val order: Ordering[String] = Ordering.String
 }
 
+/*
+ * We want to match IP address:
+ * - if equals or not equals, parse as CIDR or else IP address and work on that
+ * - if regex and other cases, parse as string
+ */
+object NodeCriterionMatcherIpaddress extends NodeCriterionMatcher {
+  private val extractor     = (n: CoreNodeFact) => n.ipAddresses.map(_.inet)
+  private val stringMatcher = NodeCriterionMatcherString(extractor)
+
+  override def matches(n: CoreNodeFact, comparator: CriterionComparator, value: String): IOResult[Boolean] = {
+    comparator match {
+      case Equals    =>
+        NodeCriterionMatcherIpaddress.buildMatcher(value, extractor(n), (ips, cidr) => ips.exists(ip => cidr.contains(ip)))
+      case NotEquals =>
+        NodeCriterionMatcherIpaddress.buildMatcher(value, extractor(n), (ips, cidr) => ips.forall(ip => !cidr.contains(ip)))
```

Comment:
equals is `exists` (at least one matches), not equals is `!forall` (non value matches)

Reply:
I find the `not exists` with the same body as for equality i.e. `!ips.exists(ip => cidr.contains(ip)))` a lot clearer than introducing another predicate `forall` here